--- a/t/test.pm	2019-06-27 16:32:17.805070422 +0000
+++ b/t/test.pm	2019-06-27 16:42:53.798767634 +0000
@@ -137,7 +137,7 @@
 use File::Path             1.0401 qw( mkpath rmtree );
 use File::Spec                0.6 qw( );
 use FindBin                  1.42 qw( $Bin );
-use POSIX                    1.02 qw( );
+use File::Temp                    qw( );
 use Test::More                    import => [qw( diag is is_deeply ok skip )];
 
 # ----------------------------------------------------------------------------
@@ -394,7 +394,7 @@
   my $tmpfh  = do { local *F; *F; };
   my $savefh = do { local *F; *F; };
 
-  my $tmpnam = POSIX::tmpnam;
+  my $tmpnam = File::Temp::tempdir;
   sysopen $tmpfh, $tmpnam, O_RDWR | O_CREAT | O_EXCL;
   unlink $tmpnam;
   select((select($tmpfh), $| = 1)[0]);
@@ -521,7 +521,7 @@
     };
 
   *tmpnam = sub {
-    my $tmpnam = POSIX::tmpnam;
+    my $tmpnam = File::Temp::tempdir;
 
     if (@_) {
       push @tmpfns, [ $tmpnam, $_[0] ];
@@ -582,7 +582,7 @@
 
 my @tmpdirs;
 sub tempdir {
-  my $tempdir = POSIX::tmpnam;
+  my $tempdir = File::Temp::tempdir;
   mkdir $tempdir, 0700
     or die "Failed to create temporary directory $tempdir: $!\n";
 
